Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1315,6 +1315,13 @@ void WebProcessProxy::didDestroyUserGestureToken(uint64_t identifier)
         m_userInitiatedActionByAuthorizationTokenMap.remove(*removed->authorizationToken());
 }
 
+void WebProcessProxy::postMessageToRemote(WebCore::ProcessIdentifier destinationProcessIdentifier, WebCore::FrameIdentifier identifier, std::optional<WebCore::SecurityOriginData> target, const WebCore::MessageWithMessagePorts& message)
```

Comment:
This applies all over the patch, but we can get the ProcessIdentifier of the frame from `frameIdentifier.processIdentifier()`

Reply:
Yeah, the issue is that the frameIdentifier doesn't have the right processIdentifier. FrameIdentifier always has the processIdentifier for the process which is created the frame object (maybe it be LocalFrame or RemoteFrame), but not which is hosting the frame.